src/main/java/org/symphonyoss/symphony/messageml/elements/TableRow.java

```diff
@@ -50,4 +68,97 @@ public void validate() throws InvalidInputException {
     assertNoText();
     assertContentModel(Arrays.asList(TableHeaderCell.class, TableCell.class));
   }
+
+  @Override
+  public Node asMarkdown() throws InvalidInputException {
+    Element parent = getParent();
+
+    if (TableBody.class.equals(parent.getClass())) {
+
+      Element grandParent = parent.getParent();
+      if (TableSelect.class.equals(grandParent.getClass())) {
+        TableSelect tableSelect = (TableSelect) grandParent;
+        return new TableSelectRowNode(tableSelect.getPosition(), tableSelect.getType(), tableSelect.getButtonText());
+      }
+
+    }
+    return new TableRowNode();
+  }
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    Element parent = getParent();
+    Element grandParent = parent.getParent();
```

Comment:
Instead of doing this Parent & Grandparent thing, why don't we actively implement new classes for the rows within a formSelect?

Reply:
In addition to what Cristiano said, why is this only parent and grandparent? A "_tr_" tag nested more than 2 levels down is treated differently than one within 2 levels of a table tag. This doesn't make sense.